src/parser.c

```
@@ -15,4 +273,6 @@ void parse(Token *tokens) {
     }
 
     sb_free(tokens);
+    
+    if(position != -1) exit(1);
```

Why explicit `exit` calls?

---

What else do I do about error exiting? There needs to be a proper error mechanism. Until then, this stays. This problem is not relevant to the parser though, so I will not put it in this PR.

---

But it's relevant to the PR, because you added the code. So you could silently fail. Or just print an error message. Point is, calling `exit` here is pretty weird, and I think we should do something else.

---

We should do something else for anything that needs an error call, not just this. I agree, there needs to be something better, but it needs to be something more than just in my code.

---

I did remove the exit call for now.